armi/bookkeeping/db/database3.py

```diff
@@ -798,6 +796,37 @@ def load(
 
         return root
 
+    def loadReadOnly(self, cycle, node, statePointName=None):
+        """Load a new reactor, in read-only mode from a DB at (cycle, node).
+
+        Parameters
+        ----------
+        cycle : int
+            Cycle number
+        node : int
+            Time node. If value is negative, will be indexed from EOC backwards like a list.
+        statePointName : str, optional
+            Statepoint name (e.g., "special" for "c00n00-special/")
+
+        Returns
+        -------
+        Reactor
+            The top-level object stored in the database; a Reactor.
+        """
+        r = self.load(cycle, node, statePointName=statePointName, allowMissing=True)
+        self._setParamsBeforeFreezing(r)
+        makeParametersReadOnly(r)
+        return r
+
+    @staticmethod
+    def _setParamsBeforeFreezing(r):
+        """Set some special case parameters before they are made read-only."""
+        for a in r.core:
```

Comment:
See comment below

Reply:
?

Reply:
Sorry, the in meant the comment about getChildren(deep=True). I wonder if we can recurse to only the last layer of the tree to call getVolume(). Similar reasoning

Reply:
Maybe this could be 

```
for obj in r.getChildren(deep=True):
    if not isinstance(obj, Component):
        continue
    obj.getVolume()  
```

or it might work to just call `r.getVolume()` since the recursion should happen all the way down the tree automatically?

Reply:
Done

Reply:
Thanks! Just out of curiosity, would `r.getVolume()` work?